db/migrate/20210715120731_fix_quarter_computation_in_reporting_tables.rb

```diff
@@ -0,0 +1,41 @@
+class FixQuarterComputationInReportingTables < ActiveRecord::Migration[5.2]
+  def up
+    drop_view :reporting_patient_states, materialized: true
+    drop_view :reporting_patient_visits, materialized: true
+    drop_view :reporting_patient_blood_pressures, materialized: true
+
+    execute "SET LOCAL TIME ZONE '#{Rails.application.config.country[:time_zone]}'"
+
+    create_view :reporting_patient_blood_pressures, version: 3, materialized: true
+    create_view :reporting_patient_visits, version: 3, materialized: true
+    create_view :reporting_patient_states, version: 3, materialized: true
+
+    add_index :reporting_patient_blood_pressures, [:month_date, :patient_id], unique: true, name: "patient_blood_pressures_patient_id_month_date"
+    add_index :reporting_patient_visits, [:month_date, :patient_id], unique: true, name: "patient_visits_patient_id_month_date"
+    add_index :reporting_patient_states, [:assigned_block_region_id], name: :patient_states_assigned_block
+    add_index :reporting_patient_states, [:assigned_district_region_id], name: :patient_states_assigned_district
+    add_index :reporting_patient_states, [:assigned_facility_region_id], name: :patient_states_assigned_facility
+    add_index :reporting_patient_states, [:assigned_state_region_id], name: :patient_states_assigned_state
+    add_index :reporting_patient_states, [:hypertension, :htn_care_state, :htn_treatment_outcome_in_last_3_months], name: :patient_states_care_state
+    add_index :reporting_patient_states, [:month_date, :patient_id], name: :patient_states_month_date_patient_id, unique: true
```

Comment:
Did you try using `update_view` here and starting w/ `patient_states` first, and working your way back?  That would mean scenic will maintain all your indexes and recreated them for you.

In any case, I suppose you can at least do `update_view` for patient_states, as nothing depends on it (yet), which would mean one less matview to have to copy/paste indexes for. 😁

Reply:
Since we're also changing the lower matviews (views etc) in this one, we have to drop `patient_states` first anyway. So we're forced to do a create instead of an update.

@harimohanraj89 in the next PR, I'd like to try using `ALTER MATERIALIZED VIEW` like I suggested [on slack](https://simpledotorg.slack.com/archives/C01JJJM1NA1/p1626369609084900).